modules/core/src/shadertools/src/modules/picking/picking.js

```diff
@@ -77,7 +77,8 @@ const float COLOR_SCALE = 1. / 255.;
  */
 vec4 picking_filterHighlightColor(vec4 color) {
   bool selected = bool(picking_vRGBcolor_Aselected.a);
-  return selected ? (picking_uHighlightColor * COLOR_SCALE) : color;
+  vec4 highlightColor = selected ? (picking_uHighlightColor * COLOR_SCALE) : color;
+  return highlightColor * 0.3 + color * 0.7;
```

Comment:
need use picking_uHighlightColor.a to blend this final color, do not hardcode the ratio like this.

Reply:
Also 1. Older behavior is as per design, if we are doing this, we should document this behavior, 2. I think we should only blend RGB and leave alpha as is, otherwise we are changing blending behavior with other layers.

Reply:
You may need use https://en.wikipedia.org/wiki/Alpha_compositing to blend the colors.

Reply:
The highlight RGB and and original RGB are blended with alpha compositing. The resulting alpha value is set the same as the original alpha value.